include/fixed_containers/reflection.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "fixed_containers/concepts.hpp"
+#include "fixed_containers/fixed_string.hpp"
+#include "fixed_containers/fixed_vector.hpp"
+
+#include <array>
+#include <cassert>
+#include <string_view>
+#include <tuple>
+
+// https://clang.llvm.org/docs/LanguageExtensions.html#builtin-dump-struct
+static_assert(__has_builtin(__builtin_dump_struct),
+              "Static reflection requires __builtin_dump_struct() to be available");
+
+namespace fixed_containers::reflection_detail
+{
+template <typename FieldTypeNameStringType = fixed_string_detail::FixedString<128>,
+          typename FieldNameStringType = fixed_string_detail::FixedString<64>>
+class FieldInfo
+{
+private:
+    FieldTypeNameStringType field_type_name_{};
+    FieldNameStringType field_name_{};
+
+public:
+    constexpr FieldInfo() noexcept
+      : field_type_name_{}
+      , field_name_{}
+    {
+    }
+
+    constexpr FieldInfo(const std::string_view& field_type_name,
+                        const std::string_view& field_name) noexcept
+      : field_type_name_{field_type_name}
+      , field_name_{field_name}
+    {
+    }
+
+public:
+    [[nodiscard]] constexpr std::string_view field_type_name() const { return field_type_name_; }
+    [[nodiscard]] constexpr std::string_view field_name() const { return field_name_; }
+};
+
+template <ConstexprDefaultConstructible T>
+constexpr std::size_t field_count_of(const T& instance = {})
+{
+    auto converter =
+        []<typename... Args>(std::size_t& out, const char* const /*fmt*/, Args&&... args)
+    {
+        if constexpr (sizeof...(args) >= 3)
```

Comment:
Is this safe to assume for future versions of clang?
Could it also pass multiple fields into one call of the converter?

Is the number of arguments still at least 3 when the field doesn't have a default value?

Reply:
> Is this safe to assume for future versions of clang?

I am looking for something better than this `>=3` check to ensure that a field is getting processed. Given the nature of `__builtin_dump_struct`, I wouldn't rule out the possibility that this will change.

> Could it also pass multiple fields into one call of the converter?

According to the documentation, the lambda is called one entry at a time (and `>=3` is currently used to deduce which ones are members).

> Is the number of arguments still at least 3 when the field doesn't have a default value?

I tested with/without default values and it doesn't make a difference - `__builtin_dump_struct` uses an instance of the struct (as opposed to just the type).